optuna/trial/_fixed.py

```diff
@@ -83,48 +87,50 @@ def suggest_float(
                     "The parameter `step` is not supported when `log` is True."
                 )
             else:
-                return self._suggest(
-                    name, distributions.DiscreteUniformDistribution(low=low, high=high, q=step)
-                )
+                return self._suggest(name, DiscreteUniformDistribution(low=low, high=high, q=step))
         else:
             if log:
-                return self._suggest(
-                    name, distributions.LogUniformDistribution(low=low, high=high)
-                )
+                return self._suggest(name, LogUniformDistribution(low=low, high=high))
             else:
-                return self._suggest(name, distributions.UniformDistribution(low=low, high=high))
+                return self._suggest(name, UniformDistribution(low=low, high=high))
 
     def suggest_uniform(self, name, low, high):
         # type: (str, float, float) -> float
 
-        return self._suggest(name, distributions.UniformDistribution(low=low, high=high))
+        return self._suggest(name, UniformDistribution(low=low, high=high))
 
     def suggest_loguniform(self, name, low, high):
         # type: (str, float, float) -> float
 
-        return self._suggest(name, distributions.LogUniformDistribution(low=low, high=high))
+        return self._suggest(name, LogUniformDistribution(low=low, high=high))
 
     def suggest_discrete_uniform(self, name: str, low: float, high: float, q: float) -> float:
-        discrete = distributions.DiscreteUniformDistribution(low=low, high=high, q=q)
+        discrete = DiscreteUniformDistribution(low=low, high=high, q=q)
         return self._suggest(name, discrete)
 
-    def suggest_int(self, name, low, high, step=1, log=False):
-        # type: (str, int, int, int, bool) -> int
-        if log:
-            sample = self._suggest(
-                name, distributions.IntLogUniformDistribution(low=low, high=high, step=step)
-            )
+    def suggest_int(self, name: str, low: int, high: int, step: int = 1, log: bool = False) -> int:
+        if step != 1:
+            if log:
+                raise ValueError(
+                    "The parameter `step != 1` is not supported when `log` is True."
+                    "The specified `step` is {}".format(step)
+                )
+            else:
+                distribution = IntUniformDistribution(
+                    low=low, high=high, step=step
+                )  # type: Union[IntUniformDistribution, IntLogUniformDistribution]
```

Comment:
Mostly curious, but is it the case that you cannot use `BaseDistribution` here?

Reply:
Actually, we can. Is it more appropriate to use `BaseDistribution`?